### PR TITLE
[fix] reading from same response body twice during prerender (#3473)

### PR DESCRIPTION
**Pull request by Theo-Steiner**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [X] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [X] This message body should clearly illustrate what problems it solves.
- [X] Ideally, include a test that fails without this PR but passes with it.

This PR fixes the behavior described in #3473, where building the default template with adapter static fails as during the pre-rendering the same response body was read twice. As @bluwy pointed out in the original issue, this is related to fetch (and accordingly node-fetch) only allowing a single read of a response body and can be circumvented by using the clone() API that is provided for this purpose.
I also wrote a test that fails without this PR, and I think that would be generally useful to have.

### Tests
- [X] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

``test/test.js:953:2 › Load › handles large responses`` fails due to problems discussed in #591

### Changesets
- [X] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/runtime/server/page/load_node.js [hunk 1]
```diff
@@ -208,7 +208,8 @@ export async function load_node({
 					const proxy = new Proxy(response, {
 						get(response, key, _receiver) {
 							async function text() {
-								const body = await response.text();
+								const clonedResponse = response.clone();
+								const body = await clonedResponse.text();
```
Comment (bluwy) [inline packages/kit/src/runtime/server/page/load_node.js:212]:
```suggestion
								const cloned_response = response.clone();
								const body = await cloned_response.text();
```

These should be snake-case for internal variables.

**Review by bluwy — COMMENTED** [review]
Looks like cloning it has an impact on large responses (probably expected) and caused the tests to fail 🤔

File: packages/kit/src/runtime/server/page/load_node.js (hunk 1 above)
Comment (Theo-Steiner) [reply packages/kit/src/runtime/server/page/load_node.js]: Fixed, Thanks! 😊

**Comment by Theo-Steiner** [comment]
> Looks like cloning it has an impact on large responses (probably expected) and caused the tests to fail 🤔

 it looks like we bumped into #591 here...

**Comment by Rich-Harris** [comment]
Thank you.

I pushed a commit that changes the behaviour so that the prerenderer gets access to the same buffered body as the code that inlines responses into the HTML (i.e. without cloning the response), though as it was on its way to GitHub I realised that this would fail in the case where the data wasn't buffered during `load`.

I guess we'll need a hybrid approach in which we use the buffered body if it's available, otherwise buffer it at prerender time. Working on it.

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**